### ISSUE TITLE
Custom Format score changed between initial grab and file import causes import to fail with "Not a Custom Format upgrade for existing episode file(s)"

### ISSUE THREAD

**Issue by HerrCraziDev** [Sonarr/Sonarr#7131]
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues

### Current Behavior

When grabbing certain episodes where the file name differs from the torrent name, the Custom Format score gets recalculated based on the file name, but only once the torrent has already been grabbed. Because of this unexpected change in CF, the newly grabbed file ends up rejected for importation with the reason "Not a Custom Format upgrade for existing episode file(s)".

### Expected Behavior

Sonarr _must_ keep the CF score consistent and _**unchanged**_ at all times during the whole episode upgrade process. It must not be recalculated based on the file name once the torrent has been grabbed. The file name inside of the torrent is outside of the user's control and ends up making the whole careful CF configuration useless by making upgrades unexpectedly fail. If the user has set up CF scores in such a way that certain releases are upgrades over others, it is not for them to be seen rejected because the _file name has a lower CF than the torrent name_ (which is a bit surprising at least). All processing should be based solely on the torrent name. 

### Steps To Reproduce

Grab any episode that has a lower CF than **an episode for which the torrent name and file name themselves have different CF scores**, then let Sonarr upgrade to the latter, only to realize the file name has a lower CF than the original file, and fail to upgrade, even tho the torrent has a higher CF than the original.

Here, the issue occurs with Oshi No Ko episodes grabbed from Nyaa, as illustrated below.

![image](https://github.com/user-attachments/assets/a395683b-d7f8-49ce-9e34-3fe264f2053c)
*fig. 1 : Episodes failing to be imported*

![image](https://github.com/user-attachments/assets/c9c0e70b-ed64-4b3b-b561-27d996406bdf)
*fig. 2 : Details of the failure, with the Erai release ending up with a lower CF*

![image](https://github.com/user-attachments/assets/29512836-f5d5-4efa-9585-02787f1da030)
*fig. 3 : Episode history, showing the higher-rated Erai-raws release being grabbed but not imported*

![image](https://github.com/user-attachments/assets/f07ec0cc-dcb0-4bbe-9bd6-46065a556d1a)
*fig. 4 : Nyaa details for the concerned episode, showing the difference between torrent title and file name*

The Erai-raws release mentioned "FR" in the title but not in the file name. On my setup, the "VOSTFR" CF has a score of 500, which is usually enough to trump non-French releases, but in this case ends up costing 500 points to the otherwise correctly-ranked Erai release, putting it lower than the file it was supposed to replace. In this case it's not a critical issue, since both releases are Crunchyroll rips with overall comparable qualities, but that's not always the case. 

### Environment

```markdown
- OS: Debian 12
- Sonarr: 4.0.8.1874
- Docker Install: No
- Using Reverse Proxy: Yes (Apache)
- Browser: Firefox
- Database: Sqlite 3.40.1
```

### What branch are you running?

Main

### Trace Logs?

-- None --

### Anything else?

[[Original episode (nyaa)](https://nyaa.si/view/1847574)] [[Newly grabbed episode (nyaa)](https://nyaa.si/view/1849467)]

**Comment by markus101** [comment]
Duplicate of https://github.com/Sonarr/Sonarr/issues/5598

**Comment by HerrCraziDev** [comment]
@markus101 Is there any existing workarounds to this issue ? 

The issue you linked to is still open and has been stale, with the discussion digressing into overcomplicated feature requests for something that should be simple. 

There seems to be no logical reason to recalculate the CF score based on the filenames in the torrent, as they are entirely out of the user's (or Sonarr's) control. This wouldn't be an issue if the CF score was recalculated based on the file name that Sonarr renamed the imported files to, as this could always be configured.
